TelCo.ColorCoder/Index.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Drawing;
+
+namespace TelCo.ColorCoder
+{
+    public interface IIndex
+    {
+        int GetIndex(Color color, bool isMajorColor=false);
+    }
+
+    public class Index: IIndex
+    {
+        public static Color[] MapMajor;
+
+        public static Color[] MapMinor;
+
+        public Index(Color[] mapMajor, Color[] mapMinor)
+        {
+            MapMajor = mapMajor;
+            MapMinor = mapMinor;
+        }
+
+        public int GetIndex(Color color, bool isMajorColor = false)
```

Comment:
GetIndex is a generic name , and if the structure stores any data that is  generic storage then this name is ok.